saw-remote-api/scripts/test_docker.sh

```diff
@@ -9,6 +9,7 @@ TAG=${1:-saw-remote-api}
 pushd $DIR/..
```

Comment:
I realize this wasn't changed in this particular PR, but in general I avoid the use of `pushd/popd` and instead use the following pattern:

```
( cd newdir;
  do_things_in_newdir;
  more_things;
)
```

There are two advantages to using this pattern instead: 1) You don't inadvertently forget a `popd` and a missing `)` will be a `bash` syntax error and more visible, and 2) the `pushd/popd` aren't always available in all shells.

I'm not necessarily indicating this should be added  to this PR, but since you're here, this might be a good time to make this change if you wanted it.

Reply:
Another recommendation is to avoid using `$DIR/..` if possible, since the `..` may move up to an unexpected location if there are links or mount points involved.  While that's unlikely in this case, `$DIR` was obtained from (effectively) `... dirname $0 ...`, so setting DIR from `dirname $(dirname $0)` might be a better approach.

Reply:
Also good suggestions. I can draft another PR to make some cleanups to Docker-related things. There are some other failures in that space that have been happening on the nightly builds, and this would fit well with those.